File: Week1_DesignPatterns/Exercise4_FinancialForecast/codes/RecursiveApproach/RecursiveForecast.cs
public class RecursiveForecast
{
    public double FutureValue(double initialValue, double growthRate, int years)
    {
        if (years == 0) //base case
        {
            return initialValue;
        }
        return (1 + growthRate) * FutureValue(initialValue, growthRate, years - 1);
    }
}

File: Week1_DesignPatterns/Exercise4_FinancialForecast/codes/RecursiveApproach/RecursiveForecast.java
public class RecursiveForecast {
    public double futureValue(double initialValue, double growthRate, int years){
        if (years == 0) {      //base case
            return initialValue;
        }
        return (1+growthRate) * futureValue(initialValue, growthRate, years-1);
    }
}
